apps/ledger-live-desktop/src/newArch/WalletSync/Flows/Synchronize/02-QRCodeStep.tsx

```diff
@@ -35,12 +43,54 @@ export default function SynchWithQRCodeStep({ displayPinCode }: Props) {
     { element: t("walletSync.synchronize.qrCode.steps.step3") },
   ];
 
-  // TO CHANGE WHEN INTRAGRATION WITH TRUSTCHAIN
+  const sdk = useTrustchainSdk();
+
+  const { trustchain, memberCredentials } = useLiveAuthenticate();
+
+  const [url, setUrl] = useState<string | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  console.log("SynchWithQRCodeStep", url);
+  console.log("error", error);
+
+  const goToActivation = useCallback(() => {
+    dispatch(setFlow({ flow: Flow.Activation, step: Step.DeviceAction }));
+  }, [dispatch]);
+
+  const onStart = useCallback(() => {
+    if (!trustchain || !memberCredentials) return;
+    setError(null);
+    createQRCodeHostInstance({
+      onDisplayQRCode: url => {
+        setUrl(url);
+      },
+      onDisplayDigits: digits => {
+        dispatch(setQrCodePinCode(digits)); // Should we assert that it is a string of 3 digits ?
```

Comment:
actually today it's "3" but it could be something else, this is a config in the protocol of QR Code, it could be changed in future and UI just need to adapt to the number of digits ideally: https://github.com/LedgerHQ/ledger-live/blob/develop/libs/trustchain/src/qrcode/index.ts#L155